src/utils/walletConnect.ts

```diff
@@ -279,25 +368,62 @@ export async function onSessionProposal(
    * probably not a big deal right now.
    */
   if (!requiredNamespaces.eip155) {
-    logger.error(new RainbowError(`WC v2: missing required namespace eip155`));
+    logger.error(new RainbowError(`WC v2: missing required namespace eip155`), {
+      requiredNamespaces,
+    });
+    return;
+  }
+
+  /**
+   * Trying to be defensive here, but I'm not sure we support this anyway so
+   * probably not a big deal right now.
+   */
+  if (!requiredNamespaces.eip155.chains) {
+    logger.error(new RainbowError(`WC v2: namespace is missing chains prop`), {
+      requiredNamespaces,
+    });
     return;
   }
 
   const { chains, methods } = requiredNamespaces.eip155;
   const chainIds = chains.map(chain => parseInt(chain.split('eip155:')[1]));
+  const unsupportedChains = chainIds.filter(id => {
+    for (const config of supportedChainConfigs) {
+      if (config.chain_id === id) return false;
+    }
+    return true;
+  });
+
+  if (unsupportedChains.length) {
+    logger.warn(`WC v2: session proposal requested unsupported chains`, {
+      unsupportedChains,
+    });
+    await rejectProposal({ proposal, reason: 'UNSUPPORTED_CHAINS' });
+    showErrorSheet({
+      reason: 'failed_wc_invalid_chains',
+      onClose() {
+        maybeGoBackAndClearHasPendingRedirect();
+      },
+    });
+    return;
+  }
+
   const peerMeta = proposer.metadata;
   const dappName =
     dappNameOverride(peerMeta.url) ||
     peerMeta.name ||
     lang.t(lang.l.walletconnect.unknown_dapp);
 
-  for (const method of methods) {
-    if (!isSigningMethod(method)) {
-      await rejectProposal({ proposal, reason: 'UNSUPPORTED_METHODS' });
-      showErrorSheet({ reason: 'failed_wc_invalid_methods' });
-      return;
-    }
-  }
+  /**
+   * Log these, but it's OK if they list them now, we'll just ignore requests
+   * to use them later.
+   */
+  const unspportedMethods = methods.filter(
+    method => !isSupportedMethod(method as RPCMethod)
+  );
+  logger.warn(`WC v2: dapp requested unsupported RPC methods`, {
+    methods: unspportedMethods,
+  });
```

Comment:
But unsupported methods are OK, we'll just reject any requests to actually use them.